documentation/src/test/java/org/hibernate/search/documentation/search/metamodel/Author__.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.documentation.search.metamodel;
+
+/**
+ * Note: this class was created by an annotation processor and copied to the sources,
+ * so that we do not run the AP on an entire documentation module.
```

Comment:
Can't you run the annotation processor on a subset of the sources?

I'm pretty sure you can add a separate execution of the maven-compiler-plugin that _only_ performs annotation processing, in which case you could possibly filter the sources to include only this package.